core/src/storage/impls/multi_version_merkle_patricia_trie/merkle_patricia_trie/cow_node_ref.rs

```diff
@@ -2,6 +2,20 @@
 // Conflux is free software and distributed under GNU General Public License.
 // See http://www.gnu.org/licenses/
 
+/// Load children merkles only when the number of uncached children nodes is
+/// above this threshold. Note that a small value will result in worse
+/// performance.
+const CHILDREN_MERKLE_UNCACHED_THRESHOLD: u32 = 8;
+
+/// Load/store children merkles only when the depth of current node is above
+/// this threshold. This is motivated by the fact that lower (deeper) nodes will
+/// be read less frequently than high nodes. The root node has depth 0. Note
+/// that a small value will result in worse performance.
+/// Depth 5 = 69905 (70k) nodes.
+/// Depth 6 = 1118481 (1.1 million) nodes.
+/// Depth 7 = 17895697 (18 million) nodes.
+const CHILDREN_MERKLE_DEPTH_THRESHOLD: u8 = 6;
```

Comment:
Depth isn't a stable measure because we do path compression when a node has only one children.. Are you talking about the number of half-bytes?

Reply:
I forgot about the compressed paths while coding it. I will redo it in the daytime.

Reply:
I think this is correct since we're caching trie nodes. The numbers only consider fully expanded paths.